pkgs/development/libraries/arcan/default.nix

```
@@ -0,0 +1,164 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, pkg-config
+, xz
+, libuvc
+, libffi
+, libX11
+, libxcb
+, libXcomposite
+, libXfixes
+, xcbutil
+, xcbutilwm
+, libXau
+, libXdmcp
+, openal
+, freetype
+, sqlite
+, libGL
+, libdrm
+, libxkbcommon
+, mesa
+, valgrind
+, luajit
+, libusb1
+, harfbuzzFull
+, SDL2
+, withVLC ? true
+, libvlc
+, withFFmpeg ? true
+, ffmpeg-full
+, withWayland ? true
+, wayland
+, wayland-protocols
+, withEspeak ? true
+, espeak-classic
+, withLibmagic ? true
+, file
+, withTesseract ? true
+, tesseract
+, withLeptonica ? true
+, leptonica
+, withVNC ? true
+, libvncserver
+, debugging ? false
+}:
+
+let
+  srcs = {
+    arcan = fetchFromGitHub {
+      owner = "letoram";
+      repo = "arcan";
+      rev = "f3341ab94b32d02f3d15c3b91a512b2614e950a5";
+      sha256 = "sha256-YBtRA5uCk4tjX3Bsu5vMkaNaCLRlM6HVQ53sna3gDsY=";
+    };
+
+    openal = fetchFromGitHub {
+      owner = "letoram";
+      repo = "openal";
+      rev = "1c7302c580964fee9ee9e1d89ff56d24f934bdef";
+      sha256 = "sha256-InqU59J0zvwJ20a7KU54xTM7d76VoOlFbtj7KbFlnTU=";
+    };
+  };
+in stdenv.mkDerivation rec {
+  pname = "arcan";
+  version = "0.6.1pre1+unstable=2021-07-08";
+
+  src = srcs.arcan;
+
+  postUnpack = ''
+    mkdir --parents $sourceRoot/external/git/openal
+    pushd $sourceRoot/external/git/
+    cp --recursive ${srcs.openal}/* openal/
+    chmod --recursive 777 openal/
```

```suggestion
    mkdir -p $sourceRoot/external/git/openal
    pushd $sourceRoot/external/git/
    cp -r ${srcs.openal}/* openal/
    chmod -R 777 openal/
```
Please use something more restrictive than 777.